### PR TITLE
BOM-1770 : Add branch parameter to make-upgrade

### PR DESCRIPTION
**Pull request by mzulqarnain1**
Relevant JIRA : https://openedx.atlassian.net/browse/BOM-1770

Related Changes in testeng-ci : https://github.com/edx/testeng-ci/pull/243

### CODE REVIEW

**Review by UsamaSadiq — APPROVED** [review]

File: testeng/jobs/upgradePythonRequirements.groovy [hunk 1]
```diff
@@ -643,7 +650,7 @@ jobConfigs.each { jobConfig ->
                     credentials('jenkins-worker')
                     url("git@github.com:${jobConfig.org}/${jobConfig.repoName}.git")
                 }
-                branch("${jobConfig.targetBranch}")
+                branch(targetBranch)
```
Comment (jmbowman) [inline testeng/jobs/upgradePythonRequirements.groovy:653]: I recall this being the part that didn't work as intended last time; you may want to model after these other jobs that check out a branch based on a parameter:

* https://github.com/edx/jenkins-job-dsl/blob/master/platform/jobs/edxEndToEndTestsJob.groovy#L186
* https://github.com/edx/jenkins-job-dsl/blob/master/platform/jobs/RetirementJobs.groovy#L152
* https://github.com/edx/jenkins-job-dsl/blob/master/testingPipelines/jobs/generalPRPipeline.groovy#L146

**Review by jmbowman — APPROVED** [review]
I think that should work; remember to revert the `bom-1770-branch` reference before merging.

**Comment by mzulqarnain1** [comment]
> I think that should work; remember to revert the `bom-1770-branch` reference before merging.

@jmbowman Done. Can you please merge both of these. As I don't have access to merge https://github.com/edx/testeng-ci/pull/243

**Comment by jmbowman** [comment]
Looks like you need to resolve some conflicts with Tim's latest PR merge first, I can merge once that's done.

**Comment by mzulqarnain1** [comment]
> Looks like you need to resolve some conflicts with Tim's latest PR merge first, I can merge once that's done.

@jmbowman Done.

**Pull request merged**